client/js/renderPreview.js

```diff
@@ -61,3 +61,56 @@ function handleImageInPreview(content, container) {
 		});
 	}
 }
+
+/* Image viewer */
+
+// FIXME Remove #input focus when this is open
+$("#viewport").on("click", ".toggle-content img", function() {
+	const self = $(this);
+	const link = self.closest("a");
+
+	openImageViewer(self, link);
+
+	// Prevent the link to open a new page since we're opening the image viewer,
+	// but keep it a link to allow for Ctrl/Cmd+click
+	event.preventDefault();
+});
+
+// Reason to make this a specific handler is, for images, to open the image
+// viewer even when clicking in the margins of the thumbnail
+$("#viewport").on("click", ".toggle-content.toggle-type-image", function() {
+	const self = $(this);
+	const image = self.find("img");
+
+	openImageViewer(image, self);
+
+	return false;
+});
+
+$(document).on("click", "#image-viewer", function() {
```

Comment:
> You can cache #viewport into a variable

Will do!

> bind all these events on it instead of document.

Any benefits? By not binding on `body`, it means your focus can be off and not catch the event, no?